### PR TITLE
[UPDATE] YUM 2 DNF, MAKE 2 CMAKE maybe later...

### PR DESCRIPTION
**Pull request by shinobu-x**
Since fedora team started to use dnf, we need to start to use that package manager. So I changed from yum to dnf used in the following shell scripts:

 run-make-check.sh
 install-deps.sh

And also because we started to use cmake for more quick-build, I separated make part to different function.

Please check!!

Those 2 scripts are really really important for someone who tries to become contributor for ceph because it's quite easy to use. Loic is really awesome!!

Thanks Guys!!
Shinobu

### CODE REVIEW

**Comment by dachary** [comment]
@shinobu-x there seems to be multiple modifications within this commit. It would help if you could split them into different commits and add commit messages that explain their rationale. Like why the installation of lsb-release should be removed etc. 

Thanks :-)

**Comment by shinobu-x** [comment]
Thank you for your quick review!!

I just want to prevent any duplication. Can I put them into just one file? Or is there any reason to keep them into two?

**Comment by dachary** [comment]
@shinobu-x duplication of what ?

File: install-deps.sh [hunk 1]
```diff
@@ -1,38 +1,29 @@
...
-
-if type zypper > /dev/null 2>&1 ; then
-    $SUDO zypper --gpg-auto-import-keys --non-interactive install openSUSE-release lsb-release
-fi
+sed -e 's/@//g' < ceph.spec.in > $DIR/ceph.spec
 
-case $(lsb_release -si) in
-Ubuntu|Debian|Devuan)
+case $DIST in
+    Ubuntu|Debian|Devuan)
```
Comment (shinobu-x) [inline install-deps.sh]: Like this line.

File: run-make-check.sh [hunk 2]
```diff
@@ -41,28 +53,45 @@ function get_processors() {
...
-    if [ -n "$install_cmd" ]; then
-        sudo $install_cmd ccache jq
+
+    if [[ `echo $DIST | grep -i fedora`  ]]; then
+        $SUDO dnf update -y
+        $SUDO dnf install -y dnf-plugins-core redhat-lsb-core ccache jq
+    elif [[ `echo $DIST | grep -i centos` ||  `echo $dist | grep -i "red hat"` ]]; then
+        $SUDO yum update -y
+        $SUDO yum install -y yum-utils redhat-lsb-core ccache jq
+    elif [[ `echo $DIST | grep -i ubuntu` ||  `echo $dist | grep -i debian` ]]; then
```
Comment (shinobu-x) [inline run-make-check.sh]: And this line.

**Comment by shinobu-x** [comment]
I think it's better to update all packages in just 1 block in 1 file. Can you get what I trying to say? Those 2 scripts don't have any magic. So they should be in 1file.

You're my hero. So it's up to you -;

File: install-deps.sh [hunk 3]
```diff
@@ -1,38 +1,29 @@
...
 # Author: Loic Dachary <loic@dachary.org>
+# Modified: Shinobu Kinjo <skinjo@redhat.com>
 #
 #  This library is free software; you can redistribute it and/or
 #  modify it under the terms of the GNU Lesser General Public
 #  License as published by the Free Software Foundation; either
 #  version 2.1 of the License, or (at your option) any later version.
 #
+LSBR=`which lsb_release`
+DIST=`$LSBR -si`
```
Comment (dachary) [inline install-deps.sh]: This will fail when lsb_release is not installed.

File: install-deps.sh [hunk 4]
```diff
@@ -1,38 +1,29 @@
...
-    $SUDO yum install -y redhat-lsb-core
-fi
-
-if type apt-get > /dev/null 2>&1 ; then
-    $SUDO apt-get install -y lsb-release
-fi
-
-if type zypper > /dev/null 2>&1 ; then
-    $SUDO zypper --gpg-auto-import-keys --non-interactive install openSUSE-release lsb-release
-fi
```
Comment (dachary) [inline install-deps.sh]: This is needed for lsb_release to be installed.

File: install-deps.sh [hunk 5]
```diff
@@ -1,38 +1,29 @@
...
+sed -e 's/@//g' < ceph.spec.in > $DIR/ceph.spec
```
Comment (dachary) [inline install-deps.sh]: There is no need to do that always, only for RPM based distributions.

File: run-make-check.sh [hunk 6]
```diff
@@ -41,28 +53,45 @@ function get_processors() {
...
+        echo "hello"
     fi
-    $DRY_RUN ./autogen.sh || return 1
-    $DRY_RUN ./configure "$@" --disable-static --with-radosgw --with-debug --without-lttng \
-        CC="ccache gcc" CXX="ccache g++" CFLAGS="-Wall -g" CXXFLAGS="-Wall -g" || return 1
-    $DRY_RUN make -j$(get_processors) || return 1
-    $DRY_RUN make $(maybe_parallel_make_check) check || return 1
-    $DRY_RUN make dist || return 1
+
+    $(do_make)
```
Comment (dachary) [inline run-make-check.sh]: do_make "$@"

File: install-deps.sh (hunk 3 above)
Comment (shinobu-x) [inline install-deps.sh]: Yes, that's my mistake. Those lines come after checking /etc/issue and installing lsb_releaes.

**Comment by shinobu-x** [comment]
I will split this PR into 2 PRs again -; This was not efficient. Sorry for that!

**Comment by dachary** [comment]
@shinobu-x you can split it into two commits maybe ? There does not seem to be a need for two PRs.

**Comment by shinobu-x** [comment]
Ah you're right. I will. Anyway I found something -; Let me update them and test them out again.
Thank you for your help!

**Comment by shinobu-x** [comment]
@dachary sorry for disturbing your vacation for such a really really unnecessary update -; when you get a chance, would you mind  checking? I've updated 2 files again. Thank you!!

**Comment by dachary** [comment]
Did you try to run install-deps.sh on a newly installed CentOS 7 with a fresh ceph clone ? I think you'll find that installing lsb-release is necessary.

**Comment by shinobu-x** [comment]
Ooooooh! Really really quick response!! I didn't though, sorry =) I noticed that you have urgent things. So you should ignore. I should be able to wait!

Anyway I will test it out with:

 CentOS
 RHEL 7

Thank you!

**Comment by shinobu-x** [comment]
Yeah it played with CentOS7, after a few changes.

   make check: successful run on ...

It's really really not quite urgent thought -;

**Comment by dachary** [comment]
install-deps.sh must run independently, not just in the context of run-make-check.sh

**Comment by shinobu-x** [comment]
@dachary I will fix it out as you mentioned. Thank you!

**Comment by dachary** [comment]
feel free to re-open when it is updated

**Pull request closed without merging**